source/gameengine/GameLogic/Joystick/SCA_Joystick.cpp

```diff
@@ -74,95 +72,82 @@ SCA_Joystick::~SCA_Joystick()
 }
 
 SCA_Joystick *SCA_Joystick::m_instance[JOYINDEX_MAX];
-int SCA_Joystick::m_joynum = 0;
-int SCA_Joystick::m_refCount = 0;
+bool SCA_Joystick::m_joystickupdatestatus = false;
```

Comment:
it was for hotpluggin but I have tested your patch and it works. Then i removed joystickupdatestatus code